functions/dailyCC/dailyCC.js

```diff
@@ -1,124 +1,89 @@
-const axios = require("axios");
-const {slugs, baseUrl} = require('./constants')
-const {randomIndex, removeFromList} = require('../utilities')
-
-const getQuestion = async (slug)=>{
-	let url = baseUrl + '/' + slug
-	try{
-		let resp = await axios.get(url)
-		return resp.data
-	}
-	catch(err){
-		throw err
-	}
-}
-
-
-const getUsedSlugs = async (firestore)=>{
-	try{
-		let snapshot = await firestore.collection('dailyCC').doc('slugs').get();
-		if(snapshot.exists){
-			let slugs = snapshot.data()
-			return slugs.usedSlugs
-		}
-		else {
-			throw 'Snapshot does not existt'
-		}
-	}
-	catch(err){
-		throw err
-	}
-
-}
-
-const getPossibleSlugs = async({firestore, ignoreList=[]})=>{
-	try{
-		let snapshot = await firestore.collection('dailyCC').doc('slugs').get();
-
-		if(snapshot.exists){
-			let slugs = snapshot.data()
-			return removeFromList(slugs.allSlugs, ignoreList)
-		}
-		else {
-			throw 'Snapshot does not existt'
-		}
+const axios = require('axios');
+const { firestore, discordMessageEmbed } = require('../config');
+const { slugs, baseUrl, codeWarsLogo } = require('./constants');
+const { randomIndex } = require('../utilities');
+
+const getQuestion = async (slug) => {
+	try {
+		const { data } = await axios.get(`${baseUrl}/${slug}`);
+		return data;
+	} catch (err) {
+		console.log({ getQuestion: err });
 	}
-	catch(err){
-		throw err
+};
+
+const getUsedSlugs = async () => {
+	try {
+		const snapshot = await firestore.collection('dailyCC').doc('slugs').get();
+		return snapshot.data().usedSlugs || [];
+	} catch (err) {
+		console.log({ getUsedSlugs: err });
 	}
+};
 
-}
-
-const addToUsedSlugs = async ({firestore, slug})=>{
-	try{
-		let usedSlugs = await getUsedSlugs(firestore)
-		usedSlugs.push(slug)
+const updateFirebaseSlugs = async (slug) => {
+	try {
+		let usedSlugs = await getUsedSlugs(firestore);
+		usedSlugs.push(slug);
 
 		await firestore
-		.collection('dailyCC')
-		.doc('slugs')
-		.set({usedSlugs}, {merge: true})
+			.collection('dailyCC')
+			.doc('slugs')
+			.set({ usedSlugs }, { merge: true });
 
-		return usedSlugs
+		return usedSlugs;
+	} catch (err) {
+		console.log({ updateFirebaseSlugs: err });
 	}
-	catch(err){
-		throw err
+};
+const embedMessage = ({ url, description, name, category, rank }) => {
+	return discordMessageEmbed
+		.setColor(rank.color.toUpperCase())
+		.setTitle(name)
+		.setDescription(`${description.substring(0, 500)}...`)
+		.setAuthor(`💡Today's Challenge`)
+		.setFooter('Codewars', codeWarsLogo)
+		.setURL(url)
+		.addFields(
+			{ name: '\u200B', value: '\u200B' },
+			{
+				name: 'Category',
+				value: category.charAt(0).toUpperCase() + category.slice(1),
+				inline: true,
+			},
+			{ name: 'Difficulty', value: rank.name, inline: true }
+		)
+		.setTimestamp();
+};
+
+const getUnusedSlug = async () => {
+	const usedSlugs = await getUsedSlugs();
+	const slug = slugs[randomIndex(slugs.length - 1)];
+	return !usedSlugs.includes(slug) ? slug : getUnusedSlug();
```

Comment:
I like the idea of using recursion, but, you aren't limiting the possible index after every loop. So essentially, the function could try the same index over and over again resulting in a quasi infinite loop.

Reply:
Just remove the usedSlugs from the list of slugs to get a definitely safe list to pull a random index from.